tests/Makefile

```diff
@@ -45,397 +45,72 @@ ifdef ZLIB
-	test_suite_cipher.des$(EXEXT)	test_suite_cipher.null$(EXEXT)	\
-	test_suite_cipher.padding$(EXEXT)				\
-	test_suite_ctr_drbg$(EXEXT)	test_suite_debug$(EXEXT)	\
-	test_suite_des$(EXEXT)		test_suite_dhm$(EXEXT)		\
-	test_suite_ecdh$(EXEXT)		test_suite_ecdsa$(EXEXT)	\
-	test_suite_ecjpake$(EXEXT)	test_suite_ecp$(EXEXT)		\
-	test_suite_error$(EXEXT)	test_suite_entropy$(EXEXT)	\
-	test_suite_gcm.aes128_de$(EXEXT)				\
-	test_suite_gcm.aes192_de$(EXEXT)				\
-	test_suite_gcm.aes256_de$(EXEXT)				\
-	test_suite_gcm.aes128_en$(EXEXT)				\
-	test_suite_gcm.aes192_en$(EXEXT)				\
-	test_suite_gcm.aes256_en$(EXEXT)				\
-	test_suite_gcm.camellia$(EXEXT)					\
-	test_suite_hmac_drbg.misc$(EXEXT)				\
-	test_suite_hmac_drbg.no_reseed$(EXEXT)				\
-	test_suite_hmac_drbg.nopr$(EXEXT)				\
-	test_suite_hmac_drbg.pr$(EXEXT)					\
-	test_suite_md$(EXEXT)		test_suite_mdx$(EXEXT)		\
-	test_suite_memory_buffer_alloc$(EXEXT)				\
-	test_suite_mpi$(EXEXT)						\
-	test_suite_pem$(EXEXT)			test_suite_pkcs1_v15$(EXEXT)	\
-	test_suite_pkcs1_v21$(EXEXT)	test_suite_pkcs5$(EXEXT)	\
-	test_suite_pkparse$(EXEXT)	test_suite_pkwrite$(EXEXT)	\
-	test_suite_pk$(EXEXT)						\
-	test_suite_rsa$(EXEXT)		test_suite_shax$(EXEXT)		\
-	test_suite_ssl$(EXEXT)		test_suite_timing$(EXEXT)			\
-	test_suite_x509parse$(EXEXT)	test_suite_x509write$(EXEXT)	\
-	test_suite_xtea$(EXEXT)		test_suite_version$(EXEXT)
+APPS =	test_suite_aes.ecb	test_suite_aes.cbc	\
+	test_suite_aes.cfb	test_suite_aes.rest	\
+	test_suite_arc4		test_suite_asn1write	\
+	test_suite_base64	test_suite_blowfish	\
+	test_suite_camellia	test_suite_ccm		\
+	test_suite_cmac						\
+	test_suite_cipher.aes					\
+	test_suite_cipher.arc4	test_suite_cipher.ccm	\
+	test_suite_cipher.gcm					\
+	test_suite_cipher.blowfish				\
+	test_suite_cipher.camellia				\
+	test_suite_cipher.des	test_suite_cipher.null	\
+	test_suite_cipher.padding				\
+	test_suite_ctr_drbg	test_suite_debug	\
+	test_suite_des		test_suite_dhm		\
+	test_suite_ecdh		test_suite_ecdsa	\
+	test_suite_ecjpake	test_suite_ecp		\
+	test_suite_error	test_suite_entropy	\
+	test_suite_gcm.aes128_de				\
+	test_suite_gcm.aes192_de				\
+	test_suite_gcm.aes256_de				\
+	test_suite_gcm.aes128_en				\
+	test_suite_gcm.aes192_en				\
+	test_suite_gcm.aes256_en				\
+	test_suite_gcm.camellia					\
+	test_suite_hmac_drbg.misc				\
+	test_suite_hmac_drbg.no_reseed				\
+	test_suite_hmac_drbg.nopr				\
+	test_suite_hmac_drbg.pr					\
+	test_suite_md		test_suite_mdx		\
+	test_suite_memory_buffer_alloc				\
+	test_suite_mpi						\
+	test_suite_pem			test_suite_pkcs1_v15	\
+	test_suite_pkcs1_v21	test_suite_pkcs5	\
+	test_suite_pkparse	test_suite_pkwrite	\
+	test_suite_pk						\
+	test_suite_rsa		test_suite_shax		\
+	test_suite_ssl		test_suite_timing			\
+	test_suite_x509parse	test_suite_x509write	\
+	test_suite_xtea		test_suite_version
 
 .SILENT:
 
 .PHONY: all check test clean
 
-all: $(APPS)
-
 $(DEP):
 	$(MAKE) -C ../library
 
 # invoke perl explicitly for the sake of mingw32-make
```

Comment:
It has been moved below https://github.com/ARMmbed/mbedtls/pull/856/files#diff-2231ca366d8aa7366f5594abfd41d66aR113
as
```
+all: $(BINARIES)
```